tests/test-definitions.sh

```diff
@@ -0,0 +1,254 @@
+    check_exit_code "IB device state: LinkUp" "IB link not up"
+}
+
+function verify_hpcx_installation {
+    check_exists "$MODULE_FILES_ROOT/mpi/hpcx"
+    
+    module load mpi/hpcx
+    local hpcx_omb_path=$MPI_HOME/tests/osu-micro-benchmarks-5.8
+    mpirun -np 2 --map-by ppr:2:node -x UCX_TLS=rc $hpcx_omb_path/osu_latency
+    check_exit_code "HPC-X $hpcx" "Failed to run HPC-X"
+    module unload mpi/hpcx
+    module purge
+}
+
+function verify_mvapich2_installation {
+    check_exists "$MODULE_FILES_ROOT/mpi/mvapich2"
+
+    module load mpi/mvapich2
+    # Env MV2_FORCE_HCA_TYPE=22 explicitly selects EDR
+    local mvapich2_omb_path=$MPI_HOME/libexec/osu-micro-benchmarks/mpi/pt2pt
+    mpiexec -np 2 -ppn 2 -env MV2_USE_SHARED_MEM=0  -env MV2_FORCE_HCA_TYPE=22  $mvapich2_omb_path/osu_latency
+    check_exit_code "MVAPICH2 $mvapich2" "Failed to run MVAPICH2"
+    module unload mpi/mvapich2
+}
+
+function verify_impi_2021_installation {
+    check_exists "$MODULE_FILES_ROOT/mpi/impi-2021"
+    
+    module load mpi/impi-2021
+    mpiexec -np 2 -ppn 2 -env FI_PROVIDER=mlx -env I_MPI_SHM=0 $MPI_BIN/IMB-MPI1 pingpong
+    check_exit_code "Intel MPI 2021 $impi_2021" "Failed to run Intel MPI 2021"
+    module unload mpi/impi-2021
+}
+
+function verify_impi_2018_installation {
+    # This needs modification
+    check_exists "$MODULE_FILES_ROOT/mpi/impi"
+
+    module load mpi/impi
+    mpiexec -np 2 -ppn 2 -env I_MPI_FABRICS=ofa ${IMPI2018_PATH}/linux/mpi/intel64/bin/IMB-MPI1 pingpong
+    check_exit_code "Intel MPI 2018: $impi_2018" "Failed to run Intel MPI 2018"
+    module unload mpi/impi
+}
+
+function verify_ompi_installation {
+    check_exists "$MODULE_FILES_ROOT/mpi/openmpi"
+    local openmpi_path=$(spack location -i openmpi@$ompi)
+    check_exists $openmpi_path
+    check_exit_code "Open MPI $ompi" "Failed to run Open MPI"
+}
+
+function verify_cuda_installation {
+    # Verify NVIDIA Driver installation
+    nvidia-smi
+    check_exit_code "Nvidia Driver $nvidia" "Failed to run Nvidia SMI"
+    
+    # Verify if NVIDIA peer memory module is inserted
+    lsmod | grep nvidia_peermem
+    check_exit_code "NVIDIA Peer memory module is inserted" "NVIDIA Peer memory module is not inserted!"
+
+    # Verify if CUDA is installed
+    nvcc --version
+    check_exit_code "CUDA Driver $cuda" "CUDA not installed"
+    check_exists "/usr/local/cuda/"
+    
+    # Verify the compilation of CUDA samples
+    /usr/local/cuda/samples/0_Introduction/mergeSort/mergeSort
+    check_exit_code "CUDA Samples $cuda" "Failed to perform merge sort using CUDA Samples"
+}
+
+function verify_nccl_installation {
+    # Print nccl.conf if it exists
+    if test -f /etc/nccl.conf; then
+        cat /etc/nccl.conf
+    fi
+
+    module load mpi/hpcx
+
+    # Check the type of Mellanox card in use
+    # mellanox_card=$(lspci -nn | grep -m 1 Mellanox | awk '{print $9 " " $10}' | sed 's/\[//' | sed 's/\]//')
+    
+    # Run NCCL test based on Mellanox card
```

Comment:
The test is based on VMSIZE not Mellanox card right?

Reply:
That is correct, it was just a placeholder